Terasology/MasterOfOreon
# Research System

Includes: 
- Construction of a new Laboratory building.
- A new Research Task which can be added by taking a book out of the Laboratory building's bookcase and placing it in the building's pedestal.
- Performing the research task. This requires the blocks mentioned in the recipe to be present in the building's chest.
- Addition of fence blocks around a area after task assignment.

Requires this https://github.com/Terasology/Books/pull/16 to be merged.

### How to test
- Select an area and assign the Laboratory construction task. The area should be resized according to the size of the building and fences be placed around this new area.
- After construction execute the command `give oreonplant0` and place the obtained items in Laboratory chest(required by the research task).
- Take the book out of bookcase(consists of the recipe which would be used for the research) and place it in the Pedestal. Oreon should perform the research task which adds a Cookie Crop block to the chest.
